### PR TITLE
Add option to scale sample image

### PR DESCRIPTION
**Pull request by IgorTatarnikov**
Before submitting a pull request (PR), please read the [contributing guide](https://github.com/brainglobe/.github/blob/main/CONTRIBUTING.md).

Please fill out as much of this template as you can, but if you have any problems or questions, just leave a comment and we will help out :)

## Description

**What is this PR**

- [ ] Bug fix
- [x] Addition of a new feature
- [ ] Other

**Why is this PR needed?**
The plugin didn't have an option to handle images that are of a different resolution to the atlas. 

**What does this PR do?**
There are now fields to enter the pixel size of the sample image and a button that downsamples it to the atlas resolution.

## References
#19 

## How has this PR been tested?
Tests have been added to cover new functionality.

## Is this a breaking change?
No.

## Does this PR require an update to the documentation?
No.

## Checklist:

- [x] The code has been tested locally
- [x] Tests have been added to cover all new functionality (unit & integration)
- [x] The code has been formatted with [pre-commit](https://pre-commit.com/)

### CODE REVIEW

File: tests/test_adjust_moving_image_view.py [hunk 1]
```diff
@@ -94,3 +100,26 @@ def test_reset_image_button_click(qtbot, adjust_moving_image_view):
     assert adjust_moving_image_view.adjust_moving_image_x.value() == 0
     assert adjust_moving_image_view.adjust_moving_image_y.value() == 0
     assert adjust_moving_image_view.adjust_moving_image_rotate.value() == 0
+
+
+@pytest.mark.parametrize(
+    "x_scale, y_scale",
+    [(2.5, 2.5), (0, 0), (10, 20), (10.222221, 10.222228)],
```
Comment (alessandrofelder) [inline tests/test_adjust_moving_image_view.py:107]: Is (0,0) actually an edge case that we should not allow, ie should the scaling factors be strictly positive and be 1 by default?

File: brainglobe_registration/registration_widget.py [hunk 2]
```diff
@@ -299,3 +303,21 @@ def _on_default_file_selection_change(
     def _on_sample_popup_about_to_show(self):
         self._sample_images = get_image_layer_names(self._viewer)
         self.get_atlas_widget.update_sample_image_names(self._sample_images)
+
+    def _on_scale_moving_image(self, x: float, y: float):
```
Comment (alessandrofelder) [inline brainglobe_registration/registration_widget.py:307]:
```suggestion
    def _on_scale_moving_image(self, x: float, y: float):
    """Scale the moving image data in napari"""
```
Or similar?

File: tests/test_registration_widget.py [hunk 3]
```diff
@@ -65,3 +65,70 @@ def test_sample_dropdown_index_changed_with_valid_index(
...
+    x_scale_factor,
+    y_scale_factor,
+):
+    mock_atlas = mocker.patch(
+        "brainglobe_registration.registration_widget.BrainGlobeAtlas"
+    )
+    mock_atlas.resolution = [20, 20, 20]
+    registration_widget._atlas = mock_atlas
+
+    curr_size = registration_widget._moving_image.data.shape
```
Comment (alessandrofelder) [inline tests/test_registration_widget.py:125]:
```suggestion
    current_size = registration_widget._moving_image.data.shape
```
(+renaming where re-used)?
Extremely nitpicky (sorry!), but all other variable names are not abbreviated.

**Review by alessandrofelder — COMMENTED** [review]
Generally code changes look good and are very clear - made some tiny comments.
Joint local testing suggests double-checking manually is a worthy next step?

File: tests/test_adjust_moving_image_view.py (hunk 1 above)
Comment (IgorTatarnikov) [reply tests/test_adjust_moving_image_view.py]: I've addressed this by setting the combo boxes to only accept values `>0.01`. If values `<= 0` somehow sneak through a warning will be shown.

File: brainglobe_registration/registration_widget.py (hunk 2 above)
Comment (IgorTatarnikov) [reply brainglobe_registration/registration_widget.py:309]: Added a doc string for this method.

**Comment by IgorTatarnikov** [comment]
I've now cached the original moving image data to allow for multiple rescaling operations to be completed. I've also disallowed pixel sizes of 0 to prevent undefined behaviour.

File: brainglobe_registration/widgets/adjust_moving_image_view.py [hunk 4]
```diff
@@ -58,9 +58,11 @@ def __init__(self, parent=None):
         rotation_range = 360
 
         self.adjust_moving_image_voxel_size_x = QDoubleSpinBox(parent=self)
-        self.adjust_moving_image_voxel_size_x.setDecimals(5)
+        self.adjust_moving_image_voxel_size_x.setDecimals(2)
+        self.adjust_moving_image_voxel_size_x.setMinimum(0.01)
```
Comment (alessandrofelder) [inline brainglobe_registration/widgets/adjust_moving_image_view.py:62]: (not sure) Should we allow the maximum [to be > the default 99.99](https://doc.qt.io/qt-6/qdoublespinbox.html#maximum-prop) (given that we have a 100um atlas)?

File: brainglobe_registration/widgets/adjust_moving_image_view.py [hunk 5]
```diff
@@ -71,13 +86,22 @@ def __init__(self, parent=None):
...
+        self.adjust_moving_image_reset_button = QPushButton()
         self.adjust_moving_image_reset_button.setText("Reset Image")
         self.adjust_moving_image_reset_button.clicked.connect(
             self._on_reset_image_button_click
         )
 
-        self.layout().addRow(QLabel("Adjust the moving image: "))
+        self.layout().addRow(QLabel("Adjust the moving image scale:"))
+        self.layout().addRow(
+            "Sample image X pixel size:", self.adjust_moving_image_voxel_size_x
```
Comment (alessandrofelder) [inline brainglobe_registration/widgets/adjust_moving_image_view.py:97]:
```suggestion
            "Sample image X pixel size [\u03BCm]:", self.adjust_moving_image_voxel_size_x
```
Worth specifying micron as unit for clarity?

**Review by alessandrofelder — APPROVED** [review]
Manual testing suggest this works nice and smoothly, and as expected, now :tada: 
Some tiny final comments to take or leave :)

File: brainglobe_registration/widgets/adjust_moving_image_view.py (hunk 4 above)
Comment (IgorTatarnikov) [reply brainglobe_registration/widgets/adjust_moving_image_view.py:62]: We can set it to be 100.00 just to catch that edge case, I don't think anyone will be acquiring anything at that coarse of a resolution but it's good to be prepared just in case.

File: brainglobe_registration/widgets/adjust_moving_image_view.py (hunk 5 above)
Comment (IgorTatarnikov) [reply brainglobe_registration/widgets/adjust_moving_image_view.py:97]: Definitely, great suggestion!
